modules/gltf/src/lib/encoders/encode-gltf.ts

```diff
@@ -19,16 +23,30 @@ export type GLTFEncodeOptions = Record<string, any>;
  * @param options
  * @returns
  */
-export function encodeGLTFSync(gltf, arrayBuffer, byteOffset, options) {
-  convertBuffersToBase64(gltf);
+export function encodeGLTFSync(
+  gltf: GLTFWithBuffers,
+  arrayBuffer: DataView | null,
+  byteOffset: number,
+  options: GLTFWriterOptions
+) {
+  let gltfToEncode = gltf;
+  if (!arrayBuffer) {
+    encodeExtensions(gltf);
+    const scenegraph = new GLTFScenegraph(gltf);
+    scenegraph.createBinaryChunk();
+    gltfToEncode = scenegraph.gltf;
+  }
+  convertBuffersToBase64(gltfToEncode);
 
   // TODO: Copy buffers to binary
 
-  return encodeGLBSync(gltf, arrayBuffer, byteOffset, options);
+  return encodeGLBSync(gltfToEncode, arrayBuffer, byteOffset, options);
 }
 
 function convertBuffersToBase64(gltf, {firstBuffer = 0} = {}) {
-  if (gltf.buffers && gltf.buffers.length > firstBuffer) {
-    throw new Error('encodeGLTF: multiple buffers not yet implemented');
+  if (gltf.buffers && gltf.buffers.length > firstBuffer + 1) {
```

Comment:
Why this logic has changed? Probably, breaking change

Reply:
Why the function name is "convertBuffersToBase64"? It doesn't convert anything, just check number of buffers

Reply:
Done.